ibm/service/vpc/resource_ibm_is_backup_policy.go

```diff
@@ -186,12 +222,25 @@ func resourceIBMIsBackupPolicyRead(context context.Context, d *schema.ResourceDa
 		log.Printf("[DEBUG] GetBackupPolicyWithContext failed %s\n%s", err, response)
 		return diag.FromErr(fmt.Errorf("[ERROR] GetBackupPolicyWithContext failed %s\n%s", err, response))
 	}
+	backupPolicy := backupPolicyIntf.(*vpcv1.BackupPolicy)
 
-	if backupPolicy.MatchResourceTypes != nil {
-		if err = d.Set("match_resource_types", backupPolicy.MatchResourceTypes); err != nil {
+	if backupPolicy.MatchResourceType != nil {
+		matchResourceTypes := *backupPolicy.MatchResourceType
+		matchResourceTypesList := []string{matchResourceTypes}
+		if err = d.Set("match_resource_types", matchResourceTypesList); err != nil {
 			return diag.FromErr(fmt.Errorf("[ERROR] Error setting match_resource_types: %s", err))
 		}
 	}
+	if backupPolicy.MatchResourceType != nil {
```

Comment:
Can't use club this if condition with the above if condition as both are same. Seems same if condition has defined two place.